flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/SpillableSubpartitionTest.java

```diff
@@ -192,21 +198,39 @@ public void testConsumeSpilledPartition() throws Exception {
 
 		Buffer read = reader.getNextBuffer();
 		assertNotNull(read);
+		assertNotSame(buffer, read);
+		assertFalse(read.isRecycled());
 		read.recycle();
+		assertTrue(read.isRecycled());
 
 		read = reader.getNextBuffer();
 		assertNotNull(read);
+		assertNotSame(buffer, read);
+		assertFalse(read.isRecycled());
 		read.recycle();
+		assertTrue(read.isRecycled());
```

Comment:
Are those `read.isRecycled()` checks somehow related to this change/fix? Or just additional side hotfix? If the latter one, I would prefer to have them in separate commit because they are confusing me.

Reply:
alright...extracted those changes into a separate hotfix